### PR TITLE
DOC: update the DataFrame.head()  docstring

### PR DESCRIPTION
**Pull request by aterrel**
- Improved summary details
- Added description of last in see-also

Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [X] PR title is "DOC: update the <your-function-or-method> docstring"
- [X] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [X] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```

################################################################################
###################### Docstring (pandas.DataFrame.head)  ######################
################################################################################

Return the first n rows.

This function returns rows for the object based on the underlying
index. It is useful for quickly testing if your object has the
right type of data in it.

Parameters
----------
n : int, default 5
    Number of rows to select.

Returns
-------
obj_head : type of caller
    The first n rows of the caller object.

See Also
--------
pandas.DataFrame.tail
    Returns the last n rows.

Examples
--------
>>> df = pd.DataFrame({'animal':['alligator', 'bee', 'falcon', 'lion',
...                    'monkey', 'parrot', 'shark', 'whale', 'zebra']})
>>> df
      animal
0  alligator
1        bee
2     falcon
3       lion
4     monkey
5     parrot
6      shark
7      whale
8      zebra

Viewing the first 5 lines

>>> df.head()
      animal
0  alligator
1        bee
2     falcon
3       lion
4     monkey

Viewing the first n lines (three in this case)

>>> df.head(3)
      animal
0  alligator
1        bee
2     falcon

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.DataFrame.head" correct. :)
```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -3607,6 +3607,10 @@ def head(self, n=5):
         """
         Return the first n rows.
 
+        This function returns rows for the object based on the underlying
```
Comment (aterrel) [inline pandas/core/generic.py]: is object the right word?

Comment (joaoavf) [inline pandas/core/generic.py]: Might be good to use 'returns the first n rows' instead of just 'returns rows'.

File: pandas/core/generic.py [hunk 2]
```diff
@@ -3607,6 +3607,10 @@ def head(self, n=5):
...
+        index. It is useful for quickly testing if your object has the
```
Comment (joaoavf) [inline pandas/core/generic.py]: Might be good to put this in a separate paragraph: 'It is useful for quickly testing if your object has the right type of data in it.'

**Review by joaoavf — COMMENTED** [review]
I could understand well the behavior of this method.

File: pandas/core/generic.py [hunk 3]
```diff
@@ -3620,6 +3624,7 @@ def head(self, n=5):
         See Also
         --------
         pandas.DataFrame.tail
```
[the remark this answers is not in the document]
Comment (aterrel) [reply pandas/core/generic.py]: Thanks for the catch!

File: pandas/core/generic.py (hunk 1 above)
Comment (aterrel) [reply pandas/core/generic.py]: Yup will update.

File: pandas/core/generic.py (hunk 2 above)
Comment (aterrel) [reply pandas/core/generic.py]: Hmm I thought that would be a bit too long.

File: pandas/core/generic.py [hunk 4]
```diff
@@ -3619,7 +3623,7 @@ def head(self, n=5):
 
         See Also
         --------
-        pandas.DataFrame.tail
+        pandas.DataFrame.tail: Returns the last n rows.
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: single backticks around `n`. Likewise on line 3622.

File: pandas/core/generic.py [hunk 5]
```diff
@@ -3607,6 +3607,10 @@ def head(self, n=5):
         """
         Return the first n rows.
 
+        This function returns the first n rows for the object based on the
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: Single backticks around `n`

File: pandas/core/generic.py [hunk 6]
```diff
@@ -3607,6 +3607,10 @@ def head(self, n=5):
         """
         Return the first n rows.
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: backticks around `n`

Comment (aterrel) [reply pandas/core/generic.py]: thanks!

**Review by jorisvandenbossche — APPROVED** [review]
Apart from the comments of Tom, looks good

File: pandas/core/generic.py (hunk 4 above)
Comment (aterrel) [reply pandas/core/generic.py]: thanks, updated.

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**